### PR TITLE
ARROW-12490: [Dev] Use only miniforge in verify-release-candidate.sh

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

**Comment by cyb70289** [comment]
I tested on `Linux/x86_64`, `Linux/aarch64`, `MacOSX/arm64`, but **not** `MacOSX/x86_64`.

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -217,17 +217,12 @@ setup_tempdir() {
...
-  elif [ "$(uname)" == "Linux" ] && [ "$(uname -m)" == "aarch64" ]; then
-    MINICONDA_URL=https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-aarch64.sh
-  else
-    MINICONDA_URL=https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
+  OS="$(uname)"
+  if [ "${OS}" == "Darwin" ]; then
+    OS=MacOSX
   fi
+  ARCH="$(uname -m)"
+  MINICONDA_URL="https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-${OS}-${ARCH}.sh"
```
Comment (cyb70289) [inline dev/release/verify-release-candidate.sh:225]: I tested on `Linux-x86_64`, `Linux-aarch64`, `MacOSX-arm64`, but **not** `MacOSX-x86_64`.

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=4.0.0 --param rc=3 verify-rc-source-*-cpp

**Comment by kszucs** [comment]
@github-actions crossbow submit --param release=4.0.0 --param rc=3 verify-rc-source-linux-cpp

**Review by kszucs — APPROVED** [review]
Thanks @cyb70289!

**Pull request closed without merging**